### PR TITLE
removal of inelastic background from empty instrumnnt

### PR DESCRIPTION
**Pull request by abuts**
**Description of work.**
The changes to the script give the possibility to remove empty background from the data runs on ISIS inelastic instruments. See the ticket #31327 for the description of the task

**To test:**

Well, I believe, there are unit tests added for every new operation. If system tests for isis direct reduction are passing, the changes will make no harm. 
The scientific validation will be performed by ISIS Excitation group. 

Fixes #31274. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by mducle — CHANGES_REQUESTED** [review]
Couple of comments, but the main thing is that we need separate background workspaces for each of the `'sample_run'`, `'wb_run'` and `'monovan_run'`.

I haven't tried running the code yet - just trying to read the code, so some of the other comments might be wrong... but some of the logic in this code is a bit convoluted.

File: scripts/Inelastic/Direct/RunDescriptor.py [hunk 1]
```diff
@@ -1496,6 +1494,104 @@ def _load_and_sum_runs(self,inst_name,monitors_with_ws):
...
+                    raise RuntimeError(
+                        'Number of spectra in background workspace (N={0}) and in Workspace {1} (N={2}) must be the same'.
+                        format(ebg_ws.getNumberHistograms(),ws.name(),ws.getNumberHistograms()))
+            #end
+
+        # Get dose for both workspaces
+        ebg_current = ebg_ws.run().getProperty('gd_prtn_chrg').value
+        ws_current = ws.run().getProperty('gd_prtn_chrg').value
+        # normalize by current and remove normalised background
+        ebg_ws  =  ebg_ws*(ws_current/ebg_current)
```
[the remark this answers is not in the document]
Comment (abuts) [reply scripts/Inelastic/Direct/RunDescriptor.py:1553]: normalization will occurs later -- the workspace after diagnostics is not normalized

File: scripts/test/RunDescriptorTest.py [hunk 2]
```diff
@@ -620,8 +621,131 @@ def test_monitors_renamed(self):
...
+        # Errors if two matrix workspaces extacted from each other are not extracted
+        ref_ws = ref_ws - 0.1*ref_ws
+
+        resWs = mtd['resWs']
+
+        difr_sample = CompareWorkspaces(ref_ws,resWs,Tolerance=1.e-7)
+        self.assertTrue(difr_sample.Result)
+
+    def test_remove_background_matrix_minus_matrix(self):
+        # Verify situation if sampe workspace is in event mode and
```
Comment (Harrietbrown) [inline scripts/test/RunDescriptorTest.py:718]:
```suggestion
        # Verify situation if sample workspace is in event mode and
```

File: scripts/test/RunDescriptorTest.py [hunk 3]
```diff
@@ -620,8 +621,131 @@ def test_monitors_renamed(self):
...
+        # and background loaded as hisograms with monitors included
```
Comment (Harrietbrown) [inline scripts/test/RunDescriptorTest.py:719]:
```suggestion
        # and background loaded as a histogram with monitors included
```

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
Could you correct these spellings.

**Review by Harrietbrown — APPROVED** [review]
This is ok to be merged pending jenkins tests passing. Good Job.

**Review by mducle — APPROVED** [review]
Looks good to me now... Thanks!

**Comment by martyngigg** [comment]
I've just realised that this I merged this to `master` without first checking where it was intended to go. There is no milestone set so is it okay that this will **not** be in 6.1?

**Comment by mducle** [comment]
@martyngigg yes I think it's ok to go into `master` not `6.1`. We're still in the process of testing it and if it's in nightly it would be easier for the instrument scientists to test it.

**Comment by abuts** [comment]
yes, Nightly would be ok

**Comment by abuts** [comment]
we will update production scripts in the normal way as we always did when Mantid acceptance cycle was too long

**Pull request merged**